### ISSUE TITLE
Cannot mount NFS synced folders

### ISSUE THREAD

**Issue by axelitus** [josenk/vagrant-vmware-esxi#54]
I cannot seem to get NFS synced folders working. I get the following error:

```
The following SSH command responded with a non-zero exit status.
Vagrant assumes that this means the command failed!

mount -o vers=3,udp 127.0.1.1:/home/axel/.homestead /vagrant

Stdout from the command:

Stderr from the command:

mount.nfs: requested NFS version or transport protocol is not supported
```

I'm guessing the problem is that instead of the 127.0.1.1 IP address it should be my network IP address. I have also tried doing the following but without luck:

```
config.vm.synced_folder('192.168.20.14:/home/axel/.homestead', '/vagrant', type: 'nfs')
```

What am I missing or is it not doable?

Both the ESXi server and my computer are on the same network.

I'm trying to get [Laravel Homestead](https://laravel.com/docs/5.6/homestead) to work through a virtual machine hosted on ESXi server instead of locally.

**Comment by axelitus** [comment]
Running `sudo mount 192.168.20.14:/home/axel/.homestead /vagrant` from within the virtual machine works, so the NFS share is configured correctly.

I'm wondering though if this is an issue with vmware_esxi provider or directly with Vagrant.

**Comment by josenk** [comment]
What version of the plugin are you using?   I fixed an issue related to this back in 2.0.6...
> https://github.com/josenk/vagrant-vmware-esxi/issues/20

**Comment by axelitus** [comment]
I have gone through the issues at [#20](https://github.com/josenk/vagrant-vmware-esxi/issues/20) and I did resolve the problem by commenting out a line in my hosts file.

As it happens ubuntu by default has an entry in the `/etc/hosts` file which points the hostname to the address 127.0.1.1 like:

```
127.0.1.1    my_host   my_host@my_domain
```

So it is a matter of DNS resolution, it resolves to that IP address instead of the network IP address.

I thought I wouldn't need to remove that line as I understand you are filtering out NFS host IP addresses that match 127.0.0.0/8 [here](https://github.com/josenk/vagrant-vmware-esxi/blob/a999bf6ae2aef3ec74491bd8eee53d1a090841f8/lib/vagrant-vmware-esxi/action/read_state.rb#L36-L39). Is there a chance those lines are not working as expected?

**Comment by josenk** [comment]
Please run this command to get the plugin version number
>  vagrant plugin list

If it's too old (you must be >2.0.6), try updating to a newer version.
>  vagrant plugin update

https://github.com/josenk/vagrant-vmware-esxi/wiki/Debugging-the-plugin
If you still have a problem, try enabling the plugin debug output.  (add the following to your Vagrantfile)
> esxi.debug = 'ip'

**Comment by axelitus** [comment]
Sorry I thought I put the versions I'm running:

OS: Ubuntu 18.04.1
Vagrant version: 2.1.2
VMware ESXI plugin version: 2.4.0

**Comment by axelitus** [comment]
I have managed to get it working "manually". I have found another issue. I'm running on a class B network (16 bit mask). The /etc/exports file does not place the entries correctly (because it does not include the mask) so the NFS share cannot be mounted.

What is expected:

```
"/home/axel/.homestead" 172.16.10.10/16(rw,no_subtree_check,all_squash,anonuid=1000,anongid=1000,fsid=3682118623)
```

What is actually placed in the file:

```
"/home/axel/.homestead" 172.16.10.10(rw,no_subtree_check,all_squash,anonuid=1000,anongid=1000,fsid=3682118623)
```

But I think this is really an issue with Vagrant and not with your plugin is it?
